src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,272 @@
+   * @param bindings The bindings that the component requires to function normally. If not set, these will be
+   * automatically resolved (with a slower execution time).
+   */
+  constructor(public element: HTMLElement, public options?: IQuerySuggestPreview, public bindings?: IComponentBindings) {
+    super(element, QuerySuggestPreview.ID, bindings);
 
     this.options = ComponentOptions.initComponentOptions(element, QuerySuggestPreview, options);
 
+    if (!this.options.resultTemplate) {
+      this.logger.warn(
+        `Option "resultTemplate" is *REQUIRED* on the component "QuerySuggestPreview". The component or the search page might *NOT WORK PROPERLY*`,
+        `Check the following URL to create your result template`,
+        `https://developers.coveo.com/x/aIGfAQ`
+      );
+    }
+
     this.bind.onRootElement(OmniboxEvents.querySuggestGetFocus, (args: IQuerySuggestSelection) => this.querySuggestGetFocus(args));
+    this.bind.onRootElement(OmniboxEvents.querySuggestRendered, () => {
+      this.handleAfterComponentInit();
+    });
+    this.bind.onRootElement(OmniboxEvents.querySuggestLooseFocus, () => {
+      this.handleQuerySuggestLooseFocus();
+    });
+  }
+
+  /**
+   * Gets the list of currently displayed result.
+   * @returns {IQueryResult[]}
+   */
+  public get displayedResults(): IQueryResult[] {
+    return this.currentlyDisplayedResults;
+  }
+
+  private get templateToHtml() {
+    const templateToHtmlArgs: ITemplateToHtml = {
+      searchInterface: this.searchInterface,
+      queryStateModel: this.queryStateModel,
+      resultTemplate: this.options.resultTemplate
+    };
+    return new TemplateToHtml(templateToHtmlArgs);
+  }
+
+  private handleQuerySuggestLooseFocus() {
+    clearTimeout(this.timer);
+    this.timer = null;
+    this.previousSuggestionHovered = null;
+    this.currentlyDisplayedResults = [];
+  }
+
+  private updatePreviewContainer(container: HTMLElement) {
+    if (!this.options.previewWidth) {
+      return;
+    }
+    container.style.width = `${this.options.previewWidth}px`;
+  }
+
+  private updateWidthOfSuggestionContainer(container: HTMLElement) {
+    if (!this.options.suggestionWidth) {
+      return;
+    }
+    container.style.minWidth = this.options.suggestionWidth;
+    container.style.maxWidth = this.options.suggestionWidth;
+  }
+
+  private handleAfterComponentInit() {
+    const suggestionContainer = $$(this.root).find('.coveo-magicbox-suggestions');
+    if (suggestionContainer) {
+      this.updateWidthOfSuggestionContainer(suggestionContainer);
+    }
+    const previewContainer = $$(this.root).find('.coveo-preview-container');
+    if (previewContainer) {
+      this.updatePreviewContainer(previewContainer);
+    }
   }
 
-  private get shouldShowPreviewResults() {
-    return this.options.numberOfPreviewResults > 0;
+  private buildPreviewHeader(suggestion: string) {
+    const text = $$('span', {}, `${this.options.headerText} "${suggestion}"`).el;
```

Comment:
Why does the text need to be inside a span ? Seems un-necessary, you can simply set it directly as text to the `header` element.

For the header, we could also be using a more semantically correct element since it's a header, like `h2` for example, instead of a div.